### PR TITLE
refactor code

### PR DESCRIPTION
**Pull request by ramsayleung**
1. refactor isSearchable to searchable 
2. stream api refactor
3. all rest-test for product-type pass

### CODE REVIEW

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetAttributeOrderService.java [hunk 1]
```diff
@@ -68,11 +71,8 @@ public void handle(ProductType entity, UpdateAction action) {
    * @return the orderd attribute names
    */
   private List<String> getOrderdAttributeNames(SetAttributeOrder action) {
-    return action.getAttributes().parallelStream()
-        .map(attribute -> {
-              return attribute.getName();
-            }
-        ).collect(Collectors.toList());
+    return action.getAttributes().parallelStream().map(AttributeDefinitionView::getName)
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/update/SetAttributeOrderService.java]: use stream to istead of parallelStream.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetAttributeOrderService.java [hunk 2]
```diff
@@ -82,10 +82,7 @@ public void handle(ProductType entity, UpdateAction action) {
    * @return the entity attribute names
    */
   private List<String> getEntityAttributeNames(ProductType entity) {
-    return entity.getAttributes().parallelStream().map(
-        attribute -> {
-          return attribute.getName();
-        }
-    ).collect(Collectors.toList());
+    return entity.getAttributes().parallelStream().map(AttributeDefinition::getName)
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/update/SetAttributeOrderService.java]: use stream to istead of parallelStream.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueOrderService.java [hunk 3]
```diff
@@ -83,12 +81,8 @@ private void setEnumValueOrder(ProductType entity, String localizedEnumAttribute
    * @return the orderd keys
    */
   private List<String> getOrderKeys(SetLocalizedEnumValueOrder setLocalizedEnumValueOrder) {
-    return setLocalizedEnumValueOrder.getValues().parallelStream()
-        .map(
-            value -> {
-              return value.getKey();
-            }
-        ).collect(Collectors.toList());
+    return setLocalizedEnumValueOrder.getValues().parallelStream().map(LocalizedEnumValue::getKey)
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueOrderService.java]: use stream to istead of parallelStream.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueOrderService.java [hunk 4]
```diff
@@ -99,23 +93,20 @@ private void setEnumValueOrder(ProductType entity, String localizedEnumAttribute
    */
   private List<String> getLocalizedEnumAttributeKeys(
       LocalizedEnumAttributeType localizedEnumAttributeType) {
-    return localizedEnumAttributeType.getValues().parallelStream().map(
-        value -> {
-          return value.getKey();
-        }
-    ).collect(Collectors.toList());
+    return localizedEnumAttributeType.getValues().parallelStream().map(LocalizedEnumValue::getKey)
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueOrderService.java]: use stream to istead of parallelStream.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetPlainEnumValueOrderService.java [hunk 5]
```diff
@@ -95,22 +98,19 @@ private void setEnumValueOrder(ProductType entity, String
    * @return the enum attribute keys
    */
   private List<String> getEnumAttributeKeys(EnumAttributeType enumType) {
-    return enumType.getValues().parallelStream().map(
-        value -> {
-          return value.getKey();
-        }
-    ).collect(Collectors.toList());
+    return enumType.getValues().parallelStream().map(EnumValue::getKey)
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/update/SetPlainEnumValueOrderService.java]: use stream to istead of parallelStream.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetPlainEnumValueLabelService.java [hunk 6]
```diff
@@ -26,19 +32,16 @@ public void handle(ProductType entity, UpdateAction action) {
...
-          }
-        }
-    );
+    Predicate<EnumValue> valuePredicate = type -> type.getKey()
+        .equals(setPlainEnumValueLabel.getNewValue().getKey());
+    Consumer<EnumValue> valueConsumer = type -> type
+        .setLabel(setPlainEnumValueLabel.getNewValue().getLabel());
+    entity.getAttributes().stream()
+        .filter(attribute -> attribute.getName().equals(setPlainEnumValueLabel.getAttributeName()))
+        .filter(attribute -> attribute.getType() instanceof EnumAttributeType).map(
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/update/SetPlainEnumValueLabelService.java]: use predicate may be better.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueOrderService.java [hunk 7]
```diff
@@ -99,23 +93,20 @@ private void setEnumValueOrder(ProductType entity, String localizedEnumAttribute
...
-   * @param entity                     the entity
+   * @param entity the entity
    * @param localizedEnumAttributeName the set plain enum value order
    * @return the enum attribute type
    */
   private LocalizedEnumAttributeType getLocalizedEnumAttributeType(ProductType entity,
-                                                                   String localizedEnumAttributeName) {
+      String localizedEnumAttributeName) {
     List<AttributeDefinition> attributes = entity.getAttributes();
     Optional<AttributeDefinition> enumAttribute = attributes.parallelStream().filter(
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueOrderService.java]: use stream to istead of parallelStream.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueLabelService.java [hunk 8]
```diff
@@ -26,19 +32,17 @@ public void handle(ProductType entity, UpdateAction action) {
...
-          }
-        }
+
+    Predicate<LocalizedEnumValue> valuePredicate = type -> type.getKey()
+        .equals(setPlainEnumValueLabel.getNewValue().getKey());
+    Consumer<LocalizedEnumValue> valueConsumer = type -> type
+        .setLabel(setPlainEnumValueLabel.getNewValue().getLabel());
+    entity.getAttributes().stream()
+        .filter(attribute -> attribute.getName().equals(setPlainEnumValueLabel.getAttributeName()))
+        .filter(attribute -> attribute.getType() instanceof LocalizedEnumAttributeType).map(
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueLabelService.java]: use predicate may be better.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetAttributeOrderService.java (hunk 1 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/update/SetAttributeOrderService.java]: refactor from parallelStream to stream.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetAttributeOrderService.java (hunk 2 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/update/SetAttributeOrderService.java]: refactor from parallelStream to stream.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueOrderService.java (hunk 3 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueOrderService.java]: refactor from parallelStream to stream.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueOrderService.java (hunk 4 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueOrderService.java]: refactor from parallelStream to stream.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueOrderService.java (hunk 7 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueOrderService.java]: refactor from parallelStream to stream.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetPlainEnumValueOrderService.java (hunk 5 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/update/SetPlainEnumValueOrderService.java]: refactor from parallelStream to stream.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetPlainEnumValueLabelService.java (hunk 6 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/update/SetPlainEnumValueLabelService.java]: extracted two filter into a predicate.

File: src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueLabelService.java (hunk 8 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/update/SetLocalizedEnumValueLabelService.java]: extracted two filter into a predicate.

**Pull request closed without merging**